### DISCUSSION TITLE
Are there any method to change the default browser to open the app?

### DISCUSSION THREAD

**Discussion by proItheus** [marimo-team/marimo#265] [Q&A] [answered]
As the title, I have a privacy hardened browser for daily web browsing, but it will break some of `marimo`'s functionality. So I'm asking sth. like jupyter's `c.NotebookApp.browser`.

**Comment by akshayka** [comment]
That's a great idea. We can add a config option to the ~/.marimo.toml for this.

**Reply by proItheus** [reply]
That's nice. BTW, what options are available for `.marimo.toml` now?

**Reply by akshayka** [reply]
We haven't made these options part of our public API yet. All of them are controlled via the settings menu in the frontend. Or at least that was true until this change, which adds a config for the browser but doesn't expose it to the frontend.

That said the available options are here for now: https://github.com/marimo-team/marimo/blob/main/marimo/_config/config.py. #269 adds an option to set the browser. We will likely make some of these options public soon; just haven't prioritized reviewing/documenting them yet.

I know there are many more options that we should probably add. Let us know if there are others you'd particularly like.

**Comment by akshayka** [comment, marked as answer]
@proItheus we've cut a new release to specify the default browser. Specify it in ~/.marimo.toml (table `[server]`, key `browser`), for example

```
[server]
browser = "firefox"
```

The supported values for `browser` are whatever Python's `webbrowser` module can handle: https://docs.python.org/3/library/webbrowser.html#webbrowser.register

**Reply by gundersona** [reply]
FYI I had issues getting this to work unless I added the directory containing the browser executable (chrome in my case) to my path env var (Windows 11).

**Reply by RiediJohannes** [reply]
I tried multiple browser values (including my default browser) as specified in the linked `webbrowser` docs and none of them worked (except "default" of course). Marimo simply raises a "webbrowser.Error: could not locate runnable browser" exception on startup.

Yes, I also tried adding the browser executable to my user (and system) env PATH and I even added the key `follow_symlink = true` in the marimo config for good measure.

Running `marimo-0.15.5` on Windows 11

**Reply by BTrey** [reply]
Is there any way to register a non-standard browser? I tried adding:

zen_path = "/home/user/.local/bin/zen"
webbrowser.register("zen", None, webbrowser.BackgroundBrowser(zen_path))

at the top of the marimo python file but that did not work.

**Reply by ozanozbeker** [reply]
On macOS I had to use:

```toml
[server]
browser = "open -a 'Google Chrome' %s"
```

I think replacing `'Google Chrome'` with any browser name should work but I haven't tested for other browsers.
